srtcore/sync.h

```diff
@@ -943,9 +943,74 @@ CUDTException& GetThreadLocalError();
 /// @param[in] maxVal maximum allowed value of the resulting random number.
 int genRandomInt(int minVal, int maxVal);
 
+class SharedMutex
+{
+    private:
+    Condition  m_pLockWriteCond;
+    Condition  m_pLockReadCond;
+
+    Mutex m_pMutex;
+
+    int m_pCountRead;
```

Comment:
"p" mean pointer, while the variable is not a pointer.
Please review the whole class.